### DISCUSSION TITLE
Next steps

### DISCUSSION THREAD

**Discussion by oskardotglobal** [winapps-org/winapps#4] [Ideas]
Moving this from #2

**Comment by LDprg** [comment]
rdp-rs seem great, because it is not just a executable we need to call, we actually could integrate the library. However I am not sure how we could only show a specific app with this library
.

**Comment by oskardotglobal** [comment]
Then the freerdp2 bindings might be our best bet

**Comment by LDprg** [comment]
Does freerdp2 has wayland support

**Reply by oskardotglobal** [reply]
No idea, the rust package are just bindings. We should be able to find that on the freerdp docs / wiki

**Reply by LDprg** [reply]
seems like xfreerdp is the command for x11 and wlfreerdp for wayland (wfreerdp is for windows) . So it propably depends on the package (freerdp-x11 or freerdp-wayland)

**Reply by oskardotglobal** [reply]
Are you experienced with rust & c? You might want to take a look at the bindings and see how the freerdp api is handling stuff

**Reply by LDprg** [reply]
I am very familiar with c & c++. In terms of rust I have some experience. I will take a look at the binding and api. 
Also a note here, we could design winapps, so we can swap freerdp with something else (just to keep possibilities open for the future).

**Reply by LDprg** [reply]
Interestingly it seems like most applications using freerdp do not use the api, instead they use the executable directly and only build a wrapper around it.

An example would be the kde rdpview:
https://invent.kde.org/network/krdc/-/blob/master/rdp/rdpview.cpp

**Reply by oskardotglobal** [reply]
I see, then maybe that would be the smartest approach
By the way, I'm not very experienced with either c++ or c

**Reply by oskardotglobal** [reply]
> Also a note here, we could design winapps, so we can swap freerdp with something else (just to keep possibilities open for the future).

That would probably the easiest approach

**Comment by LDprg** [comment]
Another thing is how do we proceed in terms of code and branches. In which branch do we add the original code base, and in which we start the rewrite.

**Reply by oskardotglobal** [reply]
I'm surprised @freechelmi didn't just fork the winapps repo, since we could've continued on main / master and rewrote on "rewrite", "v2" or something

**Reply by oskardotglobal** [reply]
We've got quite a few contributors and I think we'd be able to find more if we need to, so best to split into a rewrite and a legacy team, so whilst one team works on fixing the existing codebase and reviewing old PRs, the others can start working on the rewrite

**Comment by freechelmi** [comment]
@oskardotglobal Yes Let s move on legacy and rewrite at the same time

Please feel free to import the fmstrat main branch on develop , I did not have much time to do this bit this week

**Reply by oskardotglobal** [reply]
Ok, I'll create and populate the branches like I outlined above

**Reply by oskardotglobal** [reply]
I've created two teams which we can split into:

@winapps-org/legacy
@winapps-org/rewrite 

We also have the corresponding [`main`](https://github.com/winapps-org/winapps/tree/main) and [`rewrite`](https://github.com/winapps-org/winapps/tree/main) branches now

**Reply by freechelmi** [reply]
Great thanks !
